server/app/services/UserService.ts

```diff
@@ -64,7 +70,53 @@ export class UserService {
     if (!updatedUser) {
       throw new NotFoundError(`User with username ${username} was not found`);
     }
-
     return UserEntity.buildUser(updatedUser);
   }
+
+  public async forgotPassword(url: string, body: IUser): Promise<string> {
+    const email = body.email;
+    try {
+      const user: IUserEntity = UserEntity.buildUser(await this.userRepo.findByEmail(body));
+      const accessToken = generateToken(user);
+      requestMap.set(email, accessToken);
+
+      const info: SentMessageInfo = await this.systemService.sendEmail({
+        to: [email],
+        subject: 'ERP Password Reset',
+        emailBody: `This is the ERP Team07,\n\nPlease reset your password using the following link: ${url}/reset/${accessToken}\n\nRegards.`,
+      });
+      return info;
+    } catch (err) {
+      throw new NotFoundError(`User with email ${body.email} was not found`);
```

Comment:
this catch will be thrown if the email fails to send, either you assert on the type of error or you let the error bubble up to the controller

Reply:
its removed however I still keep the try catch for the jwt varification, having [TS7030: not all code paths return a value] by removing them.

Reply:
they are all removed